Kitodo/src/main/java/org/kitodo/production/forms/dataeditor/StructurePanel.java

```diff
@@ -1765,43 +1775,114 @@ public boolean isAssignedSeveralTimes() {
+     * The node needs to be the last node amongst its siblings.
+     * 
+     * @param node the tree node of the currently selected physical devision node
+     * @return the next logical tree node 
+     */
+    private TreeNode findNextLogicalNodeForViewAssignment(TreeNode node) {
+        if (Objects.isNull(getTreeNodeView(node))) {
+            // node is not a view
+            return null;
+        }
+
+        List<TreeNode> viewSiblings = node.getParent().getChildren();
+        if (viewSiblings.indexOf(node) != viewSiblings.size() - 1) {
+            // view is not last view amongst siblings
+            return null;
+        }
+
+        // pseudo-recursively find next logical node
+        return findNextLogicalNodeForViewAssignmentRecursive(node.getParent());
+    }
+
+    /**
+     * Find the next logical structure node that can be used to create a link by pseudo-recursively iterating over 
+     * logical parent and logical children nodes. 
+     * 
+     * @param node the tree node of the logical division
+     * @return the tree node of the next logical division
+     */
+    private TreeNode findNextLogicalNodeForViewAssignmentRecursive(TreeNode node) {
+        TreeNode current = node;
+
+        while (Objects.nonNull(current)) {
+            if (Objects.isNull(getTreeNodeStructuralElement(current))) {
+                // node is not a logical node
+                return null;
+            }
+
+            // check whether next sibling is a logical node as well
+            List<TreeNode> currentSiblings = current.getParent().getChildren();
+            int currentIndex = currentSiblings.indexOf(current);
+
+            if (currentSiblings.size() > currentIndex + 1) {
+                TreeNode nextSibling = currentSiblings.get(currentIndex + 1);
+                if (Objects.isNull(getTreeNodeStructuralElement(nextSibling))) {
+                    // next sibling is not a logical node
+                    return null;
+                }
+
+                // next sibling is a logical node and potential valid result, unless there are children
+                TreeNode nextLogical = nextSibling;
+
+                // check sibling has children (with first child being another logical node)
+                while (!nextLogical.getChildren().isEmpty()) {
+                    TreeNode firstChild = nextLogical.getChildren().get(0);
+                    if (Objects.isNull(getTreeNodeStructuralElement(firstChild))) {
+                        // first child is not a logical node
+                        return nextLogical;
+                    }
+                    // iterate to child node
+                    nextLogical = firstChild;
+                }
+                return nextLogical;
+            }
+
+            // node is last amongst siblings
+            // iterate to parent node
+            current = current.getParent();
+        }
+        return null;
+    }
```

Comment:
Method is very long and could probably be refactored in smaller methods, but as the logic is quite complex it is also good to have it in one place.